### PR TITLE
Siren Event Overhaul

### PR DESCRIPTION
**Pull request by RadiantFlash7**
## About The Pull Request
overhauls the Siren event in the various ways:

- Siren AI completely overhauled. It is now superior in most ways, and will retarget to hostiles closer to itself. it also will prevent Z level cheesing.
- Fixes Siren spawning too close with an additional check in spawn formula.
- moves siren event to /major/ ups crew requirement to 15, or 11 with significant reduced chance of activation.
- adds custom siren explosion proc to halve the damage recieved from augmentor explosions.
- lengths augmentor fire rate a bit further.
- adds in a siren boss mob,the Nemesis Assault Strider that /can/ trigger when there is 17 or more crew members on the ship.
Siren boss mob features several differant(4) attack phases.

- siren mobs in general have recieved a bit less damage from rubber rounds. Provides a time to break out lethals, and given the time and warning the crew recieves, this should be relatively reasonable.
-siren mobs not breaking machines/closets that people were hiding in, has been fixed.
- conservator laser agony reduced.
- some siren mobs have an accuracy modifier now, providing a chance for shots to deviate 15 degrees.
- This PR will add an additional 4 siren mobs:
The Monitor- A turret with a heavy damage attack but long build up, and a target indicator placed onto it's targets. very innaccurate at ranges past 7 tiles.
The Amalgamate- Ananite swarm that will engulf things and heal itself, while damaging it's victims. Able to be removed from someone with a fire extinguisher, spray of frioline, or an emp.
The Intruder- a robot with an electric overcharge ability  to increase it's movespeed and fire rate. It is equipped with a small burst laser, 
The Oracle- a teleport-based miniboss with an intangible/invulnerability ability, and two attack phases. One ranged phase that forces it's target to teleport when hit, and a melee phase that sends it's target flying a few tiles when hit.

## Why It's Good For The Game
While the siren mobs by themselves are not horribly difficult for the crew to deal with, The siren has been spawning on low pop numbers which is not ideal.
Moving the Siren to a major event, and increasing the number of spawn pods to meet it's new event level is a better solution, as is increasing the required crew to trigger.
This also overhauls the Siren AI, so it can track targets a decent ways, switch to closer targets, and prevent Z level cheese.
Also, a miniboss and boss siren mob is fun features!

## Changelog
```changelog
add: 4-5 new siren mobs
add: Siren Boss Mob
add: Siren ranged weapon inaccuracy
tweak: rewrote the AI code for siren mobs
tweak: Siren is now a major event
tweak: Siren events danger level now scales /somewhat/ based on player count a bit better.
balance: Siren mobs less damaged by rubbers
balance: damage done by augmentor explosions halved
balance: conservator laser agony reduced.
fix: siren spawns no longer spawn too close to people.
fix: siren mobs will now bash apart machines and structures containing it's target.
soundadd: added many new sounds for a siren mob
```

### CODE REVIEW

**Comment by RadiantFlash7** [comment]
Not ready yet... Pls do not look through till ready 😸

**Comment by RadiantFlash7** [comment]
Ready for review. At present, only one mob is not yet functioning 100% as intended. The Intruder's lightning AOE seems to not be providing it's 'beam' lightning sprite.

**Comment by RadiantFlash7** [comment]
I can't get the shockloop to work , but honestly it's a feature that can be added at a later date. The PR is ready to merge at this point. Pls review 😸

File: zzzz_modular_occulus/code/modules/mob/living/simple_animal/hostile/siren/intruder.dm [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+	icon_living = "intruder"
+	icon_aggro = "intruder"
+	icon_dead = "sixleg-dead"
+	icon_gib = "syndicate_gib"
+	mouse_opacity = 2
+	move_to_delay = 8
+	vision_range = 5
+	aggro_vision_range = 11
+	agony_coefficient = 0.8
+	ranged = 1
```
Comment (fludd12) [inline zzzz_modular_occulus/code/modules/mob/living/simple_animal/hostile/siren/intruder.dm:15]: Based on elsewhere, this seems to be meant as a boolean. Use TRUE for 1 and FALSE for 0

File: zzzz_modular_occulus/code/modules/mob/living/simple_animal/hostile/siren/nemesis.dm [hunk 2]
```diff
@@ -0,0 +1,280 @@
...
+	name = "nemesis assault strider"
+	desc = "A heavily armored and armed bio-mechanical beast of war. Armed with cannons and razor sharp scythe blades, very few are lucky enough to see it and survive."
+	icon = 'zzzz_modular_occulus/icons/mob/siren/nemesis.dmi'
+	icon_state = "nemesis"
+	icon_living = "nemesis"
+	icon_dead = "dead_purple"
+	icon_gib = "syndicate_gib"
+	mouse_opacity = MOUSE_OPACITY_OPAQUE
+	move_to_delay = 3
+	ranged = 1
```
Comment (fludd12) [inline zzzz_modular_occulus/code/modules/mob/living/simple_animal/hostile/siren/nemesis.dm:11]: See previous comment.

File: zzzz_modular_occulus/code/modules/mob/living/simple_animal/hostile/siren/nemesis.dm [hunk 3]
```diff
@@ -0,0 +1,280 @@
...
+	if(phase == 2)	//Plasma storm phase
+		ranged_cooldown_time = 8 SECONDS
+		ranged_cooldown = world.time + ranged_cooldown_time
+		rapid_fire_delay = 0.5 SECONDS
+		rapid = 3
+		retreat_distance = 5
+		minimum_distance = 4
+		projectiletype = /obj/item/projectile/plasma/blast
+		if(ranged != 1)
+			ranged = 1
```
Comment (fludd12) [inline zzzz_modular_occulus/code/modules/mob/living/simple_animal/hostile/siren/nemesis.dm:90]:
```suggestion
		if(!ranged)
			ranged = TRUE
```

File: zzzz_modular_occulus/code/modules/mob/living/simple_animal/hostile/siren/nemesis.dm [hunk 4]
```diff
@@ -0,0 +1,280 @@
...
+		visible_message("<span class='userdanger'>[src] primes it's heavy plasma cannons!</span>", 1)
+
+	if(phase == 3)	//Charge melee phase
+		visible_message("<span class='userdanger'>[src] lifts it's scythe-like armblades and prepares to charge!</span>", 1)
+		retreat_distance = 0
+		minimum_distance = 0
+		if(ranged == 1)
+			ranged = 0
```
Comment (fludd12) [inline zzzz_modular_occulus/code/modules/mob/living/simple_animal/hostile/siren/nemesis.dm:98]:
```suggestion
		if(ranged)
			ranged = FALSE
```

File: zzzz_modular_occulus/code/modules/mob/living/simple_animal/hostile/siren/oracle.dm [hunk 5]
```diff
@@ -0,0 +1,193 @@
...
+	icon_living = "oracle"
+	icon_aggro = "oracle"
+	icon_dead = "oracle-dead"	//turns to ash on death
+	icon_gib = "syndicate_gib"
+	mouse_opacity = 2
+	move_to_delay = 3
+	vision_range = 5
+	aggro_vision_range = 11
+	agony_coefficient = 0.6
+	ranged = 1
```
Comment (fludd12) [inline zzzz_modular_occulus/code/modules/mob/living/simple_animal/hostile/siren/oracle.dm:15]: `ranged = TRUE`

File: zzzz_modular_occulus/code/modules/mob/living/simple_animal/hostile/siren/oracle.dm [hunk 6]
```diff
@@ -0,0 +1,193 @@
...
+	sleep(2 SECONDS)
+	transcended = 0
+	visible_message("<span class='danger'>[src] rematerializes!</span>", 1)
+	icon_state = "oracle"
+
+/mob/living/simple_animal/hostile/siren/oracle/proc/phasepick()
+	set_light(3, 3, "#8B0000")
+	phase = pick(1,2)
+	if(phase == 1)	//teleport energy attack phase
+		ranged = 1
```
Comment (fludd12) [inline zzzz_modular_occulus/code/modules/mob/living/simple_animal/hostile/siren/oracle.dm:90]: `ranged = TRUE`

File: zzzz_modular_occulus/code/modules/mob/living/simple_animal/hostile/siren/oracle.dm [hunk 7]
```diff
@@ -0,0 +1,193 @@
...
+		retreat_distance = 3
+		minimum_distance = 3
+		visible_message("<span class='userdanger'>[src] primes is strange 'arm'-cannon!</span>", 1)
+
+	if(phase == 2)	//melee phase
+		ranged = 0
```
Comment (fludd12) [inline zzzz_modular_occulus/code/modules/mob/living/simple_animal/hostile/siren/oracle.dm:96]: `ranged = FALSE`

File: zzzz_modular_occulus/code/modules/mob/living/simple_animal/hostile/siren/turret.dm [hunk 8]
```diff
@@ -0,0 +1,166 @@
...
+	melee_damage_lower = 50
+	melee_damage_upper = 50
+	icon = 'zzzz_modular_occulus/icons/mob/siren/general.dmi'
+	icon_state = "passive"
+	icon_living = "passive"
+	minimum_distance = 0
+	move_to_delay = 60
+	vision_range = 12
+	aggro_vision_range = 15
+	ranged = TRUE
```
Comment (fludd12) [inline zzzz_modular_occulus/code/modules/mob/living/simple_animal/hostile/siren/turret.dm:23]: oh hey you read my mind, here 😛

File: zzzz_modular_occulus/code/modules/mob/living/simple_animal/hostile/siren/oracle.dm [hunk 9]
```diff
@@ -0,0 +1,193 @@
...
+	melee_damage_upper = 20
+	attacktext = "slammed it's energized 'arm' into"
+	environment_smash = ENVIRONMENT_SMASH_NONE
+	projectiletype = /obj/item/projectile/oracle
+	projectilesound = 'sound/weapons/energy/incinerate.ogg'
+	var/sounddelay = 0
+	var/phase_change = 15 SECONDS
+	var/phase_change_time
+	var/prob_tele = 34
+	var/transcended = 0
```
Comment (fludd12) [inline zzzz_modular_occulus/code/modules/mob/living/simple_animal/hostile/siren/oracle.dm:31]: Similarly seems to be a boolean value. `transcended = FALSE`

File: zzzz_modular_occulus/code/modules/mob/living/simple_animal/hostile/siren/oracle.dm [hunk 10]
```diff
@@ -0,0 +1,193 @@
...
+
+/mob/living/simple_animal/hostile/siren/oracle/death()
+	. = ..()
+	do_sparks(3, 0, src.loc)
+	empulse(get_turf(src), 3, 3, TRUE)
+	visible_message("<span class='danger'>[src] emits an vaguely apologetic tone and collapses into non-existance!</span>", 1)
+	qdel(src)
+
+/mob/living/simple_animal/hostile/siren/oracle/MoveToTarget()
+	if(transcended == 0 && world.time >= transcend_change_time)
```
Comment (fludd12) [inline zzzz_modular_occulus/code/modules/mob/living/simple_animal/hostile/siren/oracle.dm:62]: `if(!transcended && //etc`

File: zzzz_modular_occulus/code/modules/mob/living/simple_animal/hostile/siren/oracle.dm [hunk 11]
```diff
@@ -0,0 +1,193 @@
...
+	if(world.time >= phase_change_time)	//rotate phases ever 10 seconds
+		src.phasepick()
+		phase_change_time = world.time + phase_change
+	. = ..()
+
+/mob/living/simple_animal/hostile/siren/oracle/proc/transcend()
+	visible_message("<span class='danger'>[src] dematerializes!</span>", 1)
+	icon_state = "oracle_transition"
+	sleep(2 SECONDS)
+	transcended = 1
```
Comment (fludd12) [inline zzzz_modular_occulus/code/modules/mob/living/simple_animal/hostile/siren/oracle.dm:74]: `transcended = TRUE`

File: zzzz_modular_occulus/code/modules/mob/living/simple_animal/hostile/siren/oracle.dm [hunk 12]
```diff
@@ -0,0 +1,193 @@
...
+	icon_state = "oracle_Intangible"
+	transcend_change_time = world.time + transcend_change
+	addtimer(CALLBACK(src, .proc/descend), 3 SECONDS)
+
+/mob/living/simple_animal/hostile/siren/oracle/proc/descend()
+	icon_state = "oracle_transition"
+	sleep(2 SECONDS)
+	transcended = 0
```
Comment (fludd12) [inline zzzz_modular_occulus/code/modules/mob/living/simple_animal/hostile/siren/oracle.dm:82]: `transcended = FALSE`

File: zzzz_modular_occulus/code/modules/mob/living/simple_animal/hostile/siren/oracle.dm [hunk 13]
```diff
@@ -0,0 +1,193 @@
...
+		visible_emote("intones a strange sound!")
+		playsound(src.loc, pick(soundlist), 100, 1, 8, 8)
+		sounddelay = 30
+	else
+		sounddelay--
+
+/mob/living/simple_animal/hostile/siren/oracle/AttackingTarget()
+	if(!Adjacent(target_mob))
+		return
+	if(transcended == 1)
```
Comment (fludd12) [inline zzzz_modular_occulus/code/modules/mob/living/simple_animal/hostile/siren/oracle.dm:112]: `if(transcended)`

File: zzzz_modular_occulus/code/modules/mob/living/simple_animal/hostile/siren/oracle.dm [hunk 14]
```diff
@@ -0,0 +1,193 @@
...
+		M.attack_generic(src,rand(melee_damage_lower,melee_damage_upper),attacktext)
+		return M
+	if(istype(target_mob,/obj/machinery/bot))
+		var/obj/machinery/bot/B = target_mob
+		B.attack_generic(src,rand(melee_damage_lower,melee_damage_upper),attacktext)
+		return B
+
+
+/mob/living/simple_animal/hostile/siren/oracle/OpenFire(atom/A)
+	if(transcended == 1)
```
Comment (fludd12) [inline zzzz_modular_occulus/code/modules/mob/living/simple_animal/hostile/siren/oracle.dm:135]: `if(transcended)`

File: zzzz_modular_occulus/code/modules/mob/living/simple_animal/hostile/siren/oracle.dm [hunk 15]
```diff
@@ -0,0 +1,193 @@
...
+		if(target_mob)
+			source = target_mob
+		var/turf/T = get_random_secure_turf_in_range(source, 4, 2)
+		do_sparks(3, 0, src.loc)
+		do_teleport(src, T)
+		return FALSE
+	. = ..()
+
+/mob/living/simple_animal/hostile/siren/oracle/bullet_act(obj/item/projectile/P, def_zone)
+	if(transcended == 1)
```
Comment (fludd12) [inline zzzz_modular_occulus/code/modules/mob/living/simple_animal/hostile/siren/oracle.dm:181]: `if(transcended)` a bunch of times!

File: zzzz_modular_occulus/code/modules/mob/living/simple_animal/hostile/siren/siren.dm [hunk 16]
```diff
@@ -0,0 +1,414 @@
...
+	max_tox = 0
+	min_co2 = 0
+	max_co2 = 0
+	min_n2 = 0
+	max_n2 = 0
+	agony_coefficient = 0.5		//Gotta use lethals to put em down effectively
+	unsuitable_atoms_damage = 15
+	faction = "siren"
+	environment_smash = 1
+	destroy_surroundings = 1
```
Comment (fludd12) [inline zzzz_modular_occulus/code/modules/mob/living/simple_animal/hostile/siren/siren.dm:28]: `= TRUE`

File: zzzz_modular_occulus/code/modules/mob/living/simple_animal/hostile/siren/turret.dm [hunk 17]
```diff
@@ -0,0 +1,166 @@
...
+			if(target_dist >= 1)//Offscreen check
+				BeamStartup(target)
+				return
+
+/mob/living/simple_animal/hostile/siren/turret/proc/BeamStartup(mob/living/living_target)//It's more like requiem than final spark
+	if(combatant_state == TURRET_STATE_WARMUP && target_mob)
+		sleep(2 SECONDS)
+		update_icons()
+		living_target.add_overlay(image('zzzz_modular_occulus/icons/mob/siren/general.dmi', "targetting_indicator"))
+		targetting = 1
```
Comment (fludd12) [inline zzzz_modular_occulus/code/modules/mob/living/simple_animal/hostile/siren/turret.dm:105]: `targetting = TRUE`

File: zzzz_modular_occulus/code/modules/mob/living/simple_animal/hostile/siren/turret.dm [hunk 18]
```diff
@@ -0,0 +1,166 @@
...
+	vision_range = 12
+	aggro_vision_range = 15
+	ranged = TRUE
+	ranged_cooldown_time = 10
+	projectiletype = /obj/item/projectile/beam/sniper/siren
+	projectilesound = 'zzzz_modular_occulus/sound/effects/beamfire.ogg'
+	anchored = TRUE
+	var/combatant_state = TURRET_STATE_NEUTRAL
+	var/mob/living/beam_debuff_target
+	var/targetting = 0
```
Comment (fludd12) [inline zzzz_modular_occulus/code/modules/mob/living/simple_animal/hostile/siren/turret.dm:30]: `targetting = FALSE`
(also, it'd be _targeting_, but that's not important - there's plenty of awkward spellings throughout the code.)

File: zzzz_modular_occulus/code/modules/mob/living/simple_animal/hostile/siren/turret.dm [hunk 19]
```diff
@@ -0,0 +1,166 @@
...
+		return
+	vision_range = aggro_vision_range
+	combatant_state = TURRET_STATE_AGGRO
+	update_icons()
+	stance = HOSTILE_STANCE_ATTACK
+
+/mob/living/simple_animal/hostile/siren/turret/ForgetTarget(atom/T)
+	current_targets -= T	//removes target mob from current_targets list
+	if(T == src.target_mob)	//if mob is currently the immediate target, lose target
+		if(targetting == 1)
```
Comment (fludd12) [inline zzzz_modular_occulus/code/modules/mob/living/simple_animal/hostile/siren/turret.dm:59]: `if(targetting)`

File: zzzz_modular_occulus/code/modules/mob/living/simple_animal/hostile/siren/turret.dm [hunk 20]
```diff
@@ -0,0 +1,166 @@
...
+		living_target.add_overlay(image('zzzz_modular_occulus/icons/mob/siren/general.dmi', "targetting_indicator"))
+		targetting = 1
+		playsound(src,'zzzz_modular_occulus/sound/effects/beamcharge.ogg', 100, 0)
+		sleep(5 SECONDS)
+		set_light(3, 3, "#8B0000")
+		living_target.remove_overlays(image('zzzz_modular_occulus/icons/mob/siren/general.dmi', "targetting_indicator"))
+		combatant_state = TURRET_STATE_ACTIVE
+		living_target.update_icons()
+		living_target.add_overlay(image('zzzz_modular_occulus/icons/mob/siren/general.dmi', "targetlocked_indicator"))
+		targetting = 0
```
Comment (fludd12) [inline zzzz_modular_occulus/code/modules/mob/living/simple_animal/hostile/siren/turret.dm:113]: `targetting = FALSE`

**Review by fludd12 — CHANGES_REQUESTED** [review]
I did a more cursory review than I normally do, since I was there to watch a lot of the testing! That said, there's some code cleanup that needs to be done with booleans for readability.

**Comment by RadiantFlash7** [comment]
Requested Boolean changes made!

**Comment by RadiantFlash7** [comment]
Adjusted the balance a bit.
Removed an extraneous file as shock ability for intruder has been replaced.

File: zzzz_modular_occulus/code/game/gamemodes/events/siren_scan.dm [hunk 21]
```diff
@@ -1,18 +1,18 @@
...
 	event_type = /datum/event/siren_scan
-	event_pools = list(EVENT_LEVEL_MODERATE = POOL_THRESHOLD_MODERATE)
-	req_crew = 8
+	event_pools = list(EVENT_LEVEL_MAJOR = POOL_THRESHOLD_MAJOR*0.80)
+	req_crew = 15
 	weight = 0.5
-	max_crew_diff_lower = 3
-	tags = list(TAG_SCARY, TAG_COMMUNAL, TAG_COMBAT)
+	max_crew_diff_lower = 5
+	tags = list(TAG_SCARY, TAG_COMMUNAL, TAG_COMBAT, TAG_NEGATIVE,)
```
Comment (fludd12) [inline zzzz_modular_occulus/code/game/gamemodes/events/siren_scan.dm:15]: Trailing comma!

File: zzzz_modular_occulus/code/modules/mob/living/simple_animal/hostile/siren/oracle.dm (hunk 9 above)
Comment (fludd12) [reply zzzz_modular_occulus/code/modules/mob/living/simple_animal/hostile/siren/oracle.dm:31]: Unaddressed

File: zzzz_modular_occulus/code/modules/mob/living/simple_animal/hostile/siren/oracle.dm (hunk 10 above)
Comment (fludd12) [reply zzzz_modular_occulus/code/modules/mob/living/simple_animal/hostile/siren/oracle.dm:62]: Unaddressed

File: zzzz_modular_occulus/code/modules/mob/living/simple_animal/hostile/siren/oracle.dm (hunk 11 above)
Comment (fludd12) [reply zzzz_modular_occulus/code/modules/mob/living/simple_animal/hostile/siren/oracle.dm:74]: Unaddressed

File: zzzz_modular_occulus/code/modules/mob/living/simple_animal/hostile/siren/oracle.dm (hunk 12 above)
Comment (fludd12) [reply zzzz_modular_occulus/code/modules/mob/living/simple_animal/hostile/siren/oracle.dm:82]: Unaddressed

File: zzzz_modular_occulus/code/modules/mob/living/simple_animal/hostile/siren/oracle.dm (hunk 13 above)
Comment (fludd12) [reply zzzz_modular_occulus/code/modules/mob/living/simple_animal/hostile/siren/oracle.dm:112]: Unaddressed

File: zzzz_modular_occulus/code/modules/mob/living/simple_animal/hostile/siren/oracle.dm (hunk 14 above)
Comment (fludd12) [reply zzzz_modular_occulus/code/modules/mob/living/simple_animal/hostile/siren/oracle.dm:135]: Unaddressed

File: zzzz_modular_occulus/code/modules/mob/living/simple_animal/hostile/siren/oracle.dm (hunk 15 above)
Comment (fludd12) [reply zzzz_modular_occulus/code/modules/mob/living/simple_animal/hostile/siren/oracle.dm:181]: Unaddressed

**Review by fludd12 — CHANGES_REQUESTED** [review]
Just a few more changes, sorry!

File: zzzz_modular_occulus/code/game/gamemodes/events/siren_scan.dm (hunk 21 above)
Comment (RadiantFlash7) [reply zzzz_modular_occulus/code/game/gamemodes/events/siren_scan.dm]: Trailing comma removed,

**Comment by RadiantFlash7** [comment]
Whoops, All boolean.
Fixed up!

**Review by fludd12 — APPROVED** [review]
Looks good to me!

**Pull request merged**